packages/web-components/src/wallet-connection/watchWallet.js

```diff
@@ -79,24 +82,86 @@ export const watchWallet = (
   );
 
   const smartWalletStatusNotifierKit = makeNotifierKit(
-    /** @type { {provisioned: boolean} | null } */ (null),
+    /** @type { {provisioned: boolean, provisionFee?: NatValue} | null } */ (
+      null
+    ),
   );
 
   let lastPaths;
   let isWalletMissing = false;
+  /** @type {Promise<Tendermint34Client> | undefined} */
+  let tendermintClientP;
+
+  /** @returns {Promise<NatValue>} */
+  const fetchProvisionFee = async () => {
+    await null;
+    return new Promise((res, rej) => {
+      const query = async (attempts = 0) => {
+        await null;
+
+        try {
+          tendermintClientP ||= Tendermint34Client.connect(rpc);
+          const swingset = await querySwingsetParams(await tendermintClientP);
+          console.debug('got swingset params:', swingset);
+          const beansPerSmartWallet = swingset.params?.beansPerUnit.find(
+            ({ key }) => key === 'smartWalletProvision',
+          )?.beans;
+          const feeUnit = swingset.params?.beansPerUnit.find(
+            ({ key }) => key === 'feeUnit',
+          )?.beans;
+          const feeUnitPrice = swingset.params?.feeUnitPrice[0]?.amount;
+
+          if (!(feeUnitPrice && beansPerSmartWallet && feeUnit)) {
+            console.error('Provisioning fee missing from swingset params');
+            res(0n);
+            return;
+          }
+
+          const fee =
+            (BigInt(beansPerSmartWallet) / BigInt(feeUnit)) *
+            BigInt(feeUnitPrice);
+
+          res(fee);
+        } catch (e) {
+          console.error('Error querying smart wallet provision fee', address);
+          if (attempts >= MAX_ATTEMPTS_TO_WATCH_BANK) {
```

Comment:
retry logic seems ripe for generalization. e.g. https://caolan.github.io/async/v2/retry.js.html

or this light package https://github.com/vercel/async-retry